torch/_dynamo/variables/builtin.py

```diff
@@ -823,9 +823,21 @@ def call_dict_helper(tx, user_cls, arg, **options):
         else:
             raise AssertionError("call_dict_helper with illegal arg")
 
-    def call_dict(self, tx, obj=None):
-        if self.is_supported_call_dict_arg(tx, obj):
-            return self.call_dict_helper(tx, dict, obj)
+    def call_dict(self, tx, *args, **kwargs):
+        if not (args or kwargs):
+            return self.call_dict_helper(tx, dict, None)
+        elif (
+            not kwargs
+            and len(args) == 1
+            and self.is_supported_call_dict_arg(tx, args[0])
+        ):
+            return self.call_dict_helper(tx, dict, args[0])
+        elif not args and kwargs:
+            return variables.ConstDictVariable(
+                dict(kwargs), user_cls=dict, mutable_local=MutableLocal()
+            )
+        else:
+            unimplemented(f"dict(): {args} {kwargs}")
```

Comment:
So we now support:

1) no args or kwargs
2) single arg, no kwargs
3) no args, with kwargs

Should we make the `uimplemented` clearer that the issue is mixing both, as opposed to something in the args/kwargs? Maybe instead of dumping args/kwargs, just write a message to that effect?

Reply:
Which type of dict() constructors are you worried about here?